From f0933cc6560869b72a808896fb69789b2b38c241 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 16 Sep 2016 00:56:27 +0100
Subject: [PATCH 11/12] daemon: Tentatively address drunk OSD placement

In local testing, this appears to resolve issue #597, which we'll confirm
after more testing. Basically, the window allocation inexplicablly changes
between shows of the window, resulting in a larger offset the second time
around, moving the whole OSD one full unit east.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 daemon/osd.vala | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/daemon/osd.vala b/daemon/osd.vala
index f0f4713..13cd69f 100644
--- a/daemon/osd.vala
+++ b/daemon/osd.vala
@@ -204,14 +204,13 @@ public class OSD : Gtk.Window
         sc.get_monitor_geometry(primary_monitor, out bounds);
         Gtk.Allocation alloc;
 
-        get_allocation(out alloc);
+        get_child().get_allocation(out alloc);
 
         /* For now just center it */
         int x = bounds.x + ((bounds.width / 2) - (alloc.width / 2));
         int y = bounds.y + ((int)(bounds.height * 0.85));
         move(x, y);
     }
-
 } /* End class OSD (BudgieOSD) */
 
 /**
-- 
2.10.0

